src/neo/SmartContract/Native/Oracle/OracleRequest.cs

```diff
@@ -0,0 +1,44 @@
+using Neo.IO;
+using Neo.VM;
+using Neo.VM.Types;
+using Array = Neo.VM.Types.Array;
+
+namespace Neo.SmartContract.Native.Oracle
+{
+    public class OracleRequest : IInteroperable
+    {
+        public UInt256 OriginalTxid;
+        public long GasForResponse;
+        public string Url;
+        public string Filter;
+        public UInt160 CallbackContract;
+        public string CallbackMethod;
```

Comment:
Maybe we can force the method, and just send the url as an argument?

Reply:
The contract can use different methods to process different kind of urls.